aiokafka/coordinator/assignors/sticky/sticky_assignor.py

```diff
@@ -548,7 +552,8 @@ def _perform_reassignments(
                     )
 
                 if (
-                    partition in self.previous_assignment
+                    consumer is not None
```

Comment:
Because current_assignment is dict[str, list[TopicPartition]] and doesn't accept None key (`Invalid index type "Optional[str]" for "Dict[str, List[TopicPartition]]"; expected type "str"  [index]`). I don't see why the consumer can be None in the code. 


I didn't deep dive into the sticky assignment algorithm, but librdkafka uses assert, if `consumer` is None, see [librdkafka](https://github.com/confluentinc/librdkafka/blob/2587cac70f83fced42c51f921bed325a434f5bc7/src/rdkafka_sticky_assignor.c#L940C25-L940C34).

Reply:
Shouldn't we use `assert consumer is not None` instead of changing the branch condition?

Reply:
Done